src/librustc_typeck/check/mod.rs

```diff
@@ -5090,10 +5130,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         // provided (if any) into their appropriate spaces. We'll also report
         // errors if type parameters are provided in an inappropriate place.
 
-        let generic_segs = path_segs.iter().map(|PathSeg(_, index)| index)
-            .collect::<FxHashSet<_>>();
+        let is_alias_variant_ctor = if tcx.features().type_alias_enum_variants {
```

Comment:
Similarly to https://github.com/rust-lang/rust/pull/56225/commits/a15b1c92b38475f9d64e6418bfb604ae11d17bc9?w=1#r239266450, the feature should not change behavior, only gate it.

Reply:
You'll have to explain to me the difference. I see no conceptual difference.

Reply:
OK, if we are producing `Def::Variant` or `Def::VariantCtor` then we are doing something unstable and should report a non-fatal feature error if `features.type_alias_enum_variants` is not enabled and proceed with compilation.
So, except for the reported error behavior is identical, that's what I meant by "only gate".

In this patch `features.type_alias_enum_variants` changes behavior, e.g. successfully chooses shadowed associated items instead of variants if `features.type_alias_enum_variants` is disabled.
This means we won't be able to estimate the breakage with crater, in particular.

Reply:
Oh, I see now. Thanks for clarifying.

Reply:
(Seems like I was making life rather harder for myself than necessary!)